src/Init/Data/List/Lemmas.lean

```diff
@@ -33,6 +33,17 @@ For each `List` operation, we would like theorems describing the following, when
 
 Of course for any individual operation, not all of these will be relevant or helpful, so some judgement is required.
 
+General principles for `simp` normal forms for `List` operations:
+* Arithmetic operations are "light", so e.g. we prefer to simplify `(L.drop i).drop j` to `L.drop (i + j)`,
```

Comment:
Should this be `j + i`? Rationale: where would a constructor (`succ`) be for this to reduce. If L is headed by a cons, and `i` is headed by a `succ`, then the LHS can reduce, but the RHS not, unless you swap them around. It looks odd, but that's what we get for defining addition by recursion on the right argument …

Reply:
Yes, that's good. I better review for this problem elsewhere.

(Maybe one could feel less gross here by writing `List.drop i (List.drop j L) = List.drop (i + j) L`...)

Reply:
Oh, and in fact this is actually how the statement is. I just misquoted it in the module doc.

Reply:
I keep confusing myself always with this issue, sorry.

I guess if the proof doesn't need `add_comm` then it's right :-D

Reply:
In light of this discussion, should Length.list_append have the RHS commuted, so that `a.length` is in the “active” position?